Allow the kernel to provide random source ports.

fobser opened the pull request:
On some operating systems, for example OpenBSD since some decades, the
kernel binds to a random source port if asked for any port (port
number 0). There is no need to replicate this functionality in
userland.

ralphdolmans commented:
Hi Florian, thanks for this PR.

This option should preferably only be effective if the kernel has a proper ephemeral source port selection strategy. Since there is no detection available (and AFAIK also not easily possible), I would like to change the name of the option to something like `--disable-explicit-port-randomisation`. This matches the behavior of the option better and makes it clear you don't have port randomization if the OS doesn't do it for you.

Do you agree?

fobser committed:
Allow the kernel to provide random source ports.

On some operating systems, for example OpenBSD since some decades, the
kernel binds to a random source port if asked for any port (port
number 0). There is no need to replicate this functionality in
userland.

fobser commented:
Hi Ralph,

On Tue, Jan 07, 2020 at 01:58:50AM -0800, Ralph Dolmans wrote:
> Hi Florian, thanks for this PR.
> 
> This option should preferably only be effective if the kernel has a
> proper ephemeral source port selection strategy. Since there is no
> detection available (and AFAIK also not easily possible), I would like

Yes, I don't think it's possible to detect this. We must only disable
the explicit ramdomisation if the package maintainer requests it. Only
they know if their kernel can provide random source ports.
In all other cases we must keep the current behaviour.

> to change the name of the option to something like
> `--disable-explicit-port-randomisation`. This matches the behavior of
> the option better and makes it clear you don't have port randomization
> if the OS doesn't do it for you.
> 
> Do you agree?

Makes sense, thanks for the input, I just force-pushed an update with
this. The internet suggests that's how one does this to update a PR on
github. I hope this works out.


-- 
I'm not entirely sure you are real.

fobser commented:
Now that 1.10 is out...
I just did another force-push, Bryan Steele had pointed out that I forgot one rename in services/outside_network.h

ralphdolmans commented:
Thanks again for this. I merged the PR.
